spec/services/payment_providers/gocardless_service_spec.rb

```diff
@@ -73,4 +73,98 @@
       end
     end
   end
+
+  describe '.handle_incoming_webhook' do
+    let(:gocardless_provider) { create(:gocardless_provider, organization: organization) }
+    let(:events_result) { events['events'].map { |event| GoCardlessPro::Resources::Event.new(event) } }
+
+    let(:events) do
+      path = Rails.root.join('spec/fixtures/gocardless/events.json')
+      JSON.parse(File.read(path))
+    end
+
+    before { gocardless_provider }
+
+    it 'checks the webhook' do
+      allow(GoCardlessPro::Webhook).to receive(:parse)
+        .and_return(events_result)
+
+      result = gocardless_service.handle_incoming_webhook(
+        organization_id: organization.id,
+        body: events.to_json,
+        signature: 'signature',
+      )
+
+      expect(result).to be_success
+
+      expect(result.events).to eq(events_result)
+      expect(PaymentProviders::Gocardless::HandleEventJob).to have_been_enqueued
+    end
+
+    context 'when failing to parse payload' do
+      it 'returns an error' do
+        allow(GoCardlessPro::Webhook).to receive(:parse).and_raise(JSON::ParserError)
+
+        result = gocardless_service.handle_incoming_webhook(
+          organization_id: organization.id,
+          body: events.to_json,
+          signature: 'signature',
+        )
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::ServiceFailure)
+          expect(result.error.code).to eq('webhook_error')
+          expect(result.error.error_message).to eq('Invalid payload')
+        end
+      end
+    end
+
+    context 'when failing to validate the signature' do
+      it 'returns an error' do
+        allow(GoCardlessPro::Webhook).to receive(:parse)
+          .and_raise(GoCardlessPro::Webhook::InvalidSignatureError.new('error')
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)